dirsrvtests/tests/suites/clu/dsidm_account_test.py

```diff
@@ -120,8 +119,374 @@ def test_dsidm_account_entry_status_with_lock(topology_st, create_test_user):
+        3. The replaced description value is deleted
+    """
+
+    standalone = topology_st.standalone
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+    output = 'Successfully modified {}'.format(test_account.dn)
+
+    args = FakeArgs()
+    args.dn = test_account.dn
+    args.changes = ['add:description:new_description']
+
+    log.info('Test dsidm account modify add')
+    modify(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args, warn=False)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+    assert test_account.present('description', 'new_description')
+
+    log.info('Test dsidm account modify replace')
+    args.changes = ['replace:description:replaced_description']
+    modify(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args, warn=False)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+    assert test_account.present('description', 'replaced_description')
+    assert not test_account.present('description', 'new_description')
+
+    log.info('Test dsidm account modify delete')
+    args.changes = ['delete:description:replaced_description']
+    modify(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args, warn=False)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+    assert not test_account.present('description', 'replaced_description')
+
+
+@pytest.mark.skipif(ds_is_older("1.4.2"), reason="Not implemented")
+def test_dsidm_account_rename_by_dn(topology_st, create_test_user):
+    """ Test dsidm account rename-by-dn option
+
+    :id: f4b8e491-35b1-4113-b9c4-e0a80f8985f3
+    :setup: Standalone instance
+    :steps:
+        1. Run dsidm account rename option on existing account
+        2. Check the account does not have another uid attribute with the old rdn
+        3. Check the old account is deleted
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+
+    standalone = topology_st.standalone
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+
+    args = FakeArgs()
+    args.dn = test_account.dn
+    args.new_name = 'renamed_account'
+    args.new_dn = 'uid=renamed_account,ou=people,{}'.format(DEFAULT_SUFFIX)
+    args.keep_old_rdn = False
+
+    log.info('Test dsidm account rename-by-dn')
+    rename(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    new_account = accounts.get(args.new_name)
+    output = 'Successfully renamed to {}'.format(new_account.dn)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+
+    log.info('Verify the new account does not have a uid attribute with the old rdn')
+    assert not new_account.present('uid', test_user_name)
+    assert new_account.present('displayName', test_user_name)
+
+    log.info('Verify the old account does not exist')
+    assert not test_account.exists()
+
+    log.info('Clean up')
+    new_account.delete()
```

Comment:
Oh, and also, please put the clean-up code into `try-finally` wherever possible.

Reply:
Fixed